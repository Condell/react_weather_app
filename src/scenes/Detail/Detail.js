import React from "react";
import styled from "styled-components";

import WeatherIcon from "../../components/WeatherIcon/WeatherIcon";

const DetailWrapper = styled.div`
  font-family: 'Abel', sans-serif;
  text-align: center;
`;

const Detail = (props, match) =>
  <DetailWrapper>
    <WeatherIcon
      large
      src={`http://openweathermap.org/img/w/${props.location.state.weather[0]
        .icon}.png`}
      alt={`${props.location.state.weather[0].description} icon`}
    />
    <h2>
      {props.location.state.date}
    </h2>
    <h2>
      {props.match.params.cityName}
    </h2>
    <h2>
      Forecast: {props.location.state.weather[0].description}
    </h2>
    <h2>
      Low Temperature: {Math.round(props.location.state.temp.min)} °F
    </h2>
    <h2>
      High Temperature: {Math.round(props.location.state.temp.max)} °F
    </h2>
    <h2>
      Humidity:{" "}
      {props.location.state.humidity === 0
        ? "Unlisted"
        : props.location.state.humidity}
    </h2>
  </DetailWrapper>;

export default Detail;
