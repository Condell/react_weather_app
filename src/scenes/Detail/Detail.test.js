import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import Detail from "./Detail";

describe("Detail", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Detail />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});