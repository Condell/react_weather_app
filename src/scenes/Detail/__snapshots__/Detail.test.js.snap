// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail should render 1`] = `
<styled.div>
  <h1>
    DETAILS
  </h1>
</styled.div>
`;
